opal-core-api/src/main/java/org/obiba/opal/core/support/vcf/VCFSamplesSummaryBuilder.java

```diff
@@ -0,0 +1,135 @@
+package org.obiba.opal.core.support.vcf;
+
+import com.google.common.base.Strings;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Sets;
+import org.obiba.magma.Datasource;
+import org.obiba.magma.ValueSet;
+import org.obiba.magma.ValueTable;
+import org.obiba.magma.Variable;
+import org.obiba.opal.core.domain.VCFSamplesMapping;
+import org.obiba.opal.spi.vcf.VCFStore;
+
+import java.util.Collection;
+import java.util.Set;
+
+public class VCFSamplesSummaryBuilder {
+
+  public VCFSamplesSummaryBuilder() {
+  }
+
+  private VCFSamplesMapping mappings;
+  private VCFStore store;
+  private Datasource datasource;
+
+  public VCFSamplesSummaryBuilder datasource(Datasource value) {
+    datasource = value;
+    return this;
+  }
+
+  public VCFSamplesSummaryBuilder mappings(VCFSamplesMapping value) {
+    mappings = value;
+    return this;
+  }
+
+  public VCFSamplesSummaryBuilder store(VCFStore value) {
+    store = value;
+    return this;
+  }
+
+  public Stats buildSummary() {
+    Stats stats = new Stats();
+
+    if (mappings != null) {
+      ValueTable vt = datasource.getValueTable(mappings.getTableName());
```

Comment:
Datasource not needed, use MagmaEngineTableResolver instead to get a ValueTable from its fully qualified name.